src/codegen/generators/expression_generator.rs

```diff
@@ -745,6 +750,58 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
         .map(Into::into)
     }
 
+    pub fn generate_variadic_arguments_list(
+        &self,
+        pou: &PouIndexEntry,
+        variadic_params: &[&AstStatement],
+    ) -> Result<Vec<BasicValueEnum<'ink>>, Diagnostic> {
+        let generated_params = variadic_params
+            .iter()
+            .map(|param_statement| {
+                self.get_type_hint_for(param_statement)
+                    .map(|it| it.get_name())
+                    .and_then(|type_name| self.generate_argument_by_val(type_name, param_statement))
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        //get the real varargs from the index
+        if let Some(VarArgs::Sized(Some(type_name))) = self
+            .index
+            .get_variadic_member(pou.get_name())
+            .and_then(|it| it.get_varargs())
+        {
+            let llvm_type = if pou.is_generic() {
+                //Use the type of the first parameter
+                let type_info = self.get_type_hint_info_for(variadic_params[0])?;
+                self.llvm_index.get_associated_type(type_info.get_name())
+            } else {
+                self.llvm_index.get_associated_type(type_name)
+            }?;
+            let size = generated_params.len();
+            let size_param = self
+                .llvm_index
```

Comment:
we already expose i32_type in llvm.rs, 
I suggest we also expose size_type() that gives you the plattform's default intType